src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -106,11 +123,14 @@ private void validateDatasetName(String datasetName, Errors errors) {
   }
 
   private void validateDatePartitionOptions(
-      DatePartitionOptionsModel options, List<ColumnModel> columns, Errors errors) {
+      DatePartitionOptionsModel options,
+      List<ColumnModel> columns,
+      Errors errors,
+      SchemaValidationContext context) {
```

Comment:
Now that I see how many methods `context` needs to be added to, I think it may be good to rethink how this validator works. The reason why `DatasetRequestValidator` is injectable is so web APIs can use its `validate()` method. But there are cases where we also want to validate things outside of this code path; in those cases it would be better to avoid creating `DatasetRequestValidator` since those cases don't call `validate()` and only call other methods in this class.

I would change it like this
- move all the common validation code into its own class. To do this I'd move these methods into `SchemaValidationContext`, then rename it something else and move it to a top-level class. It's inside `DatasetRequestValidator` but it doesn't need any of its state.
- replace all cases where the code is injecting `DatasetRequestValidator` directly to instead use this new class

Reply:
Okay, I made these changes, but decided to keep the name as it does still provide schema validation context, thoughts?